beacon-chain/state/stategen/getter.go

```diff
@@ -62,16 +62,18 @@ func (s *State) StateByRoot(ctx context.Context, blockRoot [32]byte) (state.Beac
 
 // StateByRootInitialSync retrieves the state from the DB for the initial syncing phase.
 // It assumes initial syncing using a block list rather than a block tree hence the returned
-// state is not copied.
-// It invalidates cache for parent root because pre state will get mutated.
-// Do not use this method for anything other than initial syncing purpose or block tree is applied.
+// state is not copied (block batches returned from initial sync are linear).
+// It invalidates cache for parent root because pre-state will get mutated.
+//
+// WARNING: Do not use this method for anything other than initial syncing purpose or block tree is applied.
```

Comment:
What does it mean "or block tree is applied"?

Reply:
@nisdas , can you help me out with a better comment?